### PR TITLE
Packaging

### PR DESCRIPTION
**Pull request by timmens**
### What problem do you want to solve?

Automate PyPI publishing using GitHub actions. This is a preparation step before tackling #502.

### Todo

- [x] Automate PyPI publishing using workflow
   - [x] Set up workflow
   - [x] Create secret token `PYPI_API_TOKEN` (@hmgaudecker, I cannot do this since token creation requires access to the repository settings)
      From [pypi-publish-notes](https://github.com/marketplace/actions/pypi-publish):
      > The secret used in ${{ secrets.PYPI_API_TOKEN }} needs to be created on the settings page of your project on GitHub. See [Creating & using secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,7 +39,6 @@ waf-*
 waf3-*
 *latexmk
 *synctex.gz
-.*
 !.environment.osx.yml
```
Comment (hmgaudecker) [inline .gitignore:42]:
```suggestion
.
!.github
```
Please continue to ignore hidden files and whitelist those that should be included under VC.

The whitelist patterns for the environment files are from a very distant past and can all be removed...

**Comment by hmgaudecker** [comment]
So I registered on PyPI and created the token. However, I don't think I can *register the project* ? Seems like that will happen with the upload? Or I do not find the option...

**Comment by hmgaudecker** [comment]
I am pretty sure that `setup.py` is not required anymore; remember having had that discussion with @tobiasraabe, probably in the context of #450, but I cannot check ATM.

**Comment by timmens** [comment]
After discussing with @janosg we came to a conclusion that
```yaml
on: push
```
is preferred to
```yaml
on: release
```
While the latter specifies more directly that we want to publish to PyPI only when creating a new release, we realized that the former also tests that in each action run the package building works.

What remains to be discussed @hmgaudecker

1. I suggest merging this PR and testing it directly by releasing a new patch on main. I've tried the workflow on one of my repositories, and it works.
2. I would also suggest updating the documentation in a different PR. The reason is that I don't know all the details of the steps that have to be taken when releasing a new version on conda-forge yet, since I can only start working on this component once the project has been uploaded to PyPI.

**Comment by timmens** [comment]
I've just tried it out on one of my projects. The workflow is triggered when releasing a new version from a different branch than main, and the package is uploaded to PyPI. So yes, I strongly expect that this works.

**Review by hmgaudecker — APPROVED** [review]
Great, let's give it a shot!

**Pull request merged**